manager/logbroker/broker.go

```diff
@@ -288,23 +335,47 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) error {
 		return err
 	}
 
+	var currentSubscription *subscription
+
+	// Make *sure* to call `done()` before returning from this function,
+	// otherwise the clients will be stuck on waiting for the stream to
+	// terminate.
+	done := func(err error) error {
```

Comment:
Yeah but then I'd need to store the error locally in order to return it and I find it error prone, e.g. `err = blah() ; return err` everywhere in the code